src/dmd/typesem.d

```diff
@@ -1336,6 +1336,29 @@ extern(C++) Type typeSemantic(Type type, const ref Loc loc, Scope* sc)
 
                 Type t = fparam.type.toBasetype();
 
+                if (sc.flags & SCOPE.Cfile)
+                {
+                    /* C11 6.7.6.3-7 array of T is converted to pointer to T
+                     */
+                    if (auto ta = t.isTypeDArray())
+                    {
+                        t = ta.next.pointerTo();
+                        fparam.type = t;
```

Comment:
What about implicit conversion for callers? Don't they need to be handled explicitly too?

Reply:
Yes, but it's completely different. This PR is about adjusting the *type* of the function parameter. You're talking about doing a *conversion* of an expression. It's covered by different paragraphs in the spec. I plan on the next PR being about that.